tobias/tools/submerge.py

```diff
@@ -0,0 +1,164 @@
+                    case 5:
+                        splitline.append('.')            # add a strand placeholder
+
+            # if strand was not specified, intersect with both
+            if splitline[5] != '+' and splitline[5] != '-':
+
+                splitline[5] = '+'
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+                splitline[5] = '-'
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+            else:
+                line_string = "\t".join(splitline)
+                checked_regions.append(line_string)
+
+    # write checked regions to file
+    logger.debug("Writing checked regions to file")
+    regions = os.path.join(outdir, "regions.tmp")
+    with open(regions, "w") as f:
+        for line in checked_regions:
+            f.write(line + "\n")
+
+    # Read TFs
+    if args.TF is not None:
+        logger.debug("Reading TFs from: " + args.TF)
+        tfs = pd.read_csv(args.TF, header=None, names=["TF"])
+        tfs = tfs["TF"].tolist()
+
+        # get only those TFs that were provided
+        tfbs_files = [tfbs for tfbs in tfbs_files if os.path.basename(os.path.dirname(tfbs)) in tfs]
+
+    # intersect each TFBS file with the regions
+    logger.debug('Intersecting query with all TBFS files')
+    all_intersections = ''
+    for file in tfbs_files:
+        tf = os.path.basename(os.path.dirname(file))
+
+        # remove header
+        headless_file = os.path.join(outdir, f"{tf}.headless")
+        command = f'sed "1d" {file} > {headless_file}'
+        subprocess.run(command, shell=True)
+
+        # intersect
+        command = f"bedtools intersect -a {regions} -b {headless_file} -wa -wb"
+        intersection = subprocess.check_output(command, shell=True).decode("utf-8")
+        all_intersections += intersection
+
+        # remove headless file
+        os.remove(headless_file)
+
+    # make list out of lines
+    all_intersections = all_intersections.strip().split("\n")
+
+    # get tfbs header
+    with open(tfbs_files[0]) as f:
+        tfbs_header = f.readline()
+
+    # merge all intersection files
+    logger.debug('Merging all intersections')
+    with open(args.output, "w") as f:
+
+        if args.output.endswith(".bed"):
+            pass
+        else:
+            query_header = "query chr\tquery start\tquery end\tquery id\tquery score\tquery strand\t"
+            header = query_header + tfbs_header
+            f.write(header)
+
+        for line in all_intersections:
+            f.write(line + "\n")
+
+    # filter output file
+    df = pd.read_csv(args.output, sep="\t")
+
+    logger.debug('Sorting')
+
+    # remove 'chr' str from all chr columns TODO not optimal as contigs may be named differently, i.e. 'contig' or 'chrIV'
+    df["query chr"] = df["query chr"].str.replace("chr", "")
+    df["TFBS_chr"] = df["TFBS_chr"].str.replace("chr", "")
```

Comment:
If you do this for sorting purposes alone consider using the [`key` parameter](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.sort_values.html) instead.